src/main/java/org/kiwiproject/jersey/client/ServiceIdentifier.java

```diff
@@ -2,46 +2,46 @@
 
 import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotBlank;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.util.Duration;
 import lombok.Builder;
-import lombok.Getter;
+import lombok.Value;
 import lombok.With;
+import org.kiwiproject.registry.model.Port;
 
-import java.time.Duration;
 import java.util.Optional;
 
 /**
  * Service definition used for the client connection
  */
-@Getter
+@Value
 public class ServiceIdentifier {
-    // TODO: Should we just use service-discovery-client's PortType?
-    public enum Connector {
-        ADMIN, APPLICATION
-    }
 
     @With
-    private final String serviceName;
+    String serviceName;
 
-    private final String preferredVersion;
-    private final String minimumVersion;
-    private final Connector connector;
-    private final Duration connectTimeout;
-    private final Duration readTimeout;
+    String preferredVersion;
+    String minimumVersion;
+    Port.PortType connector;
+    Duration connectTimeout;
+    Duration readTimeout;
 
+    @JsonCreator
     @Builder(toBuilder = true)
-    public ServiceIdentifier(String serviceName,
-                             String preferredVersion,
-                             String minimumVersion,
-                             Connector connector,
-                             Duration connectTimeout,
-                             Duration readTimeout) {
+    public ServiceIdentifier(@JsonProperty("serviceName") String serviceName,
```

Comment:
Do we want this to be public? I _think_ Jackson can still de-serialize if not public. Of course, if and when we go to the module system that might not be the case (not sure)

Reply:
Good point, I'll make it private